nowriter/tmdecoding/tm_decoder_pure.go

```diff
@@ -0,0 +1,241 @@
+	const primitiveSize int = 2
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = int16(binary.BigEndian.Uint16(inp[:primitiveSize]))
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeInt32(inp []byte) (i int32, bytesRead int, err error) {
+	const primitiveSize int = 4
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = int32(binary.BigEndian.Uint32(inp[:primitiveSize]))
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeInt64(inp []byte) (i int64, bytesRead int, err error) {
+	const primitiveSize int = 8
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = int64(binary.BigEndian.Uint64(inp[:primitiveSize]))
+	bytesRead = primitiveSize
+	return
+}
+func (e TMDecoderPure) DecodeOctet(inp []byte) (b byte, bytesRead int, err error) {
+	const primitiveSize int = 1
+	if len(inp) < primitiveSize {
+		return
+	}
+	b = inp[0]
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeOctets(inp []byte) (b []byte, bytesRead int, err error) {
+	bytesRead = len(inp)
+	if bytesRead > 0 {
+		b = make([]byte, bytesRead)
+		copy(b, inp)
+	}
+	return
+}
+
+func (e TMDecoderPure) DecodeTime(inp []byte) (t time.Time, bytesRead int, err error) {
+	const primitiveSize int = 8
+	if len(inp) < primitiveSize {
+		return
+	}
+	i := int64(binary.BigEndian.Uint64(inp[:primitiveSize]))
+	t = time.Unix(0, i)
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeUint8(inp []byte) (i uint8, bytesRead int, err error) {
+	const primitiveSize int = 1
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = uint8(inp[0])
+	bytesRead = primitiveSize
+	return
+}
+func (e TMDecoderPure) DecodeUint16(inp []byte) (i uint16, bytesRead int, err error) {
+	const primitiveSize int = 2
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = uint16(binary.BigEndian.Uint16(inp[:primitiveSize]))
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeUint16s(inp []byte) (iz []uint16, bytesRead int, err error) {
+	if len(inp) < 4 {
+		return
+	}
+	count := int(uint32(binary.BigEndian.Uint32(inp[:4])))
+	wantedBytes := int(4 + 2*count)
+	if len(inp) < wantedBytes {
+		return
+	}
+	iz = make([]uint16, count)
+	bytesRead = wantedBytes
+	for i := 0; i < count; i += 1 {
+		iz[i] = uint16(binary.BigEndian.Uint16(inp[2*i : 2*i+1]))
+	}
+	return
+}
+
+func (e TMDecoderPure) DecodeUint32(inp []byte) (i uint32, bytesRead int, err error) {
+	const primitiveSize int = 4
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = uint32(binary.BigEndian.Uint32(inp[:primitiveSize]))
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeUint64(inp []byte) (i uint64, bytesRead int, err error) {
+	const primitiveSize int = 8
+	if len(inp) < primitiveSize {
+		return
+	}
+	i = uint64(binary.BigEndian.Uint64(inp[:primitiveSize]))
+	bytesRead = primitiveSize
+	return
+}
+
+func (e TMDecoderPure) DecodeUvarint(inp []byte) (i uint, bytesRead int, err error) {
+	inpSize := len(inp)
+	if len(inp) < 1 {
```

Comment:
`inpSize`

Reply:
thanks @melekes i believe i just pushed a commit that fixes this everywhere